judgels-backends/jophiel/jophiel-app/src/main/java/judgels/jophiel/session/SessionConfiguration.java

```diff
@@ -0,0 +1,23 @@
+package judgels.jophiel.session;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import java.util.Optional;
+import org.immutables.value.Value;
+
+@Value.Immutable
+@JsonDeserialize(as = ImmutableSessionConfiguration.class)
+public interface SessionConfiguration {
+    judgels.jophiel.session.SessionConfiguration DEFAULT = new SessionConfiguration.Builder()
+            .maxConcurrentSessionsPerUser(-1)
+            .disableLogOut(false)
+            .build();
+
+    // If maxConcurrentSessionsPerUser is not set or negative, concurrent session limitation is disabled.
+    // This limitation does not apply for admins.
+    Optional<Integer> getMaxConcurrentSessionsPerUser();
```

Comment:
Can we use `@Value.Default` instead of `Optional`? Because we need to check for -1 anyway.